File: shared/domain/models/notification.go
package models

import "time"

// NotificationType は通知の種別。
type NotificationType string

const (
	NotificationTypeInvitation    NotificationType = "invitation"     // 任命招待
	NotificationTypeLending       NotificationType = "lending"        // 区域の貸し出し
	NotificationTypeReturn        NotificationType = "return"         // 返却
	NotificationTypeForceReturn   NotificationType = "force_return"   // 強制回収
	NotificationTypeRequestResult NotificationType = "request_result" // 申請結果
)

// Notification は通知を表す。
type Notification struct {
	ID          string           `json:"id"`
	Type        NotificationType `json:"type"`
	TargetID    string           `json:"targetId"`    // 宛先DID
	ReferenceID string           `json:"referenceId"` // 関連エンティティID
	Message     string           `json:"message"`
	Read        bool             `json:"read"`
	CreatedAt   time.Time        `json:"createdAt"`
	ExpiresAt   *time.Time       `json:"expiresAt"` // 表示期限
}

File: pwa/src/domain/models/notification.ts
// 通知のドメインモデル。
// 仕様: docs/wants/07_通知と申請.md
// 参照実装: shared/domain/models/notification.go

/** 通知の種別。 */
export type NotificationType =
  | "invitation" // 任命招待
  | "lending" // 区域の貸し出し
  | "area_invite" // 区域招待（被招待者向け、発行時のみ）
  | "return" // 返却
  | "force_return" // 強制回収
  | "request_result"; // 申請結果

export interface Notification {
  id: string;
  type: NotificationType;
  /** 宛先DID */
  targetId: string;
  /** 関連エンティティID */
  referenceId: string;
  message: string;
  read: boolean;
  /** ISO 8601 */
  createdAt: string;
  /** 表示期限（ISO 8601） */
  expiresAt: string | null;
}
